app/services/manage_action.rb

```diff
@@ -33,11 +31,18 @@ def queue_message
   end
 
   def previous_action
-    Action.where(action_user: action_user, page_id: page).first
+    @previous_action ||= Action.where(action_user: action_user, page_id: page).first
   end
 
   def action_user
-    @user ||= ActionUser.find_or_create_by(email: @params[:email])
+    return @user if @user.present?
+    @user = ActionUser.find_or_create_by(email: @params[:email])
+    @params.each_pair do |key, value|
+      # here we choose not to overwrite newly blank values.
+      @user[key] = value if @user.respond_to?("#{key}=".to_sym) && value.present?
+    end
```

Comment:
The form can have lots of fields that are submitted to AK that cannot be stored on the ActionUser (eg `is_delta_shareholder`). the controller filters to just the form fields, but I can't use strong params outside the controller. however, I would like to make this more succinct...